HW1/P4/P4.txt

```diff
@@ -0,0 +1,14 @@
+# Your discussion here
+
+1. for the graph representation, firstly convert the initial RDD into a RDD (issue, {all characters in the issue}). Then, convert the new RDD into a RDD (node, {neighbor1, neighbor2, ...}) whose values' type are all set. 
+
+2. the number of touched nodes:(including the start node itself)
+"CAPTAIN AMERICA": 6048
+"MISS THING/MARY": 7
+"ORWELL":9
+
+If a node doesn't have a defined distance, there will be no path between the start node and this node, which means the node's character has never appeared in the same comic issues as the character of the start node or of its neighbors. For example, the touched nodes for "CAPTAIN AMERICA" is 6048 and the total nodes number is 6444. Thus, only a few characters have never been in the same issues as "CAPTAIN AMERICA" or as its neighbors. 
```

Comment:
Not just its neighbors, though, right? A node without a defined distance/no path between the nodes means that the characters has never appeared in the same issue as Captain America, any neighbors, neighbors of those neighbors, and so on indefinitely - in graph theoretical terms, it means that node is on a different connected component (shades of P5...)